### PR TITLE
Describe how window mode setter and getter works

### LINKED ISSUE #53414: `OS.window_fullscreen` does not exist any more
### Godot version

v4.0.dev.custom_build [2e8cba0bd]

### System information

Linux Mint - Vulkan API 1.2.131 - Using Vulkan Device #0: NVIDIA - NVIDIA GeForce GTX 1080

### Issue description

Changing fullscreen mode via `OS.window_fullscreen = !OS.window_fullscreen` does not work any more and will result in the following error:

```
Invalid get index 'window_fullscreen' (on base: 'OS).
```

### Steps to reproduce

Put this code in any script on any node and press `F11` while the game is running
```
extends Node2D

func _input(event):
	if event.is_echo():
		return
		
	if event is InputEventKey:
		if event.keycode == KEY_F11:
			if event.is_pressed():
				OS.window_fullscreen = !OS.window_fullscreen
```

Using this line instead also does not work

```
				OS.set_deferred("window_fullscreen", true)
```

### Minimal reproduction project

_No response_

### PR DESCRIPTION
**Pull request by DeeJayLSP**
*Bugsquad edit:* Fixes #53414.

### CODE REVIEW

File: doc/classes/DisplayServer.xml [hunk 1]
```diff
@@ -666,6 +667,26 @@
...
 			<argument index="1" name="window_id" type="int" default="0" />
 			<description>
+				Sets window mode for the given window to [code]mode[/code].
+				[codeblocks]
+				[gdscript]
+				# Set window mode to:
+				DisplayServer.window_set_mode(0) # Windowed
+				DisplayServer.window_set_mode(1) # Minimized
+				DisplayServer.window_set_mode(2) # Maximized
+				DisplayServer.window_set_mode(3) # Fullscreen
```
Comment (Calinou) [inline doc/classes/DisplayServer.xml:677]: We don't need a code sample here, as usage becomes obvious as soon as you look at the WindowMode constant.

**Review by Calinou — DISMISSED** [review]

File: doc/classes/DisplayServer.xml (hunk 1 above)
Comment (DeeJayLSP) [reply doc/classes/DisplayServer.xml]: I didn't realize at first that you could link the enum, done.

File: doc/classes/DisplayServer.xml [hunk 2]
```diff
@@ -666,6 +667,11 @@
 			<argument index="0" name="mode" type="int" enum="DisplayServer.WindowMode" />
 			<argument index="1" name="window_id" type="int" default="0" />
 			<description>
+				Sets window mode for the given window to [code]mode[/code].
+				See [enum DisplayServer.WindowMode] for possible values and how each mode behaves.
+				
+				[b]Note:[/b] Setting the window to full screen forcibly sets the borderless flag to [code]true[/code], so make sure to set it back to [code]false[/code] when not wanted.
+
```
Comment (akien-mga) [inline doc/classes/DisplayServer.xml:674]:
```suggestion
				Sets window mode for the given window to [code]mode[/code]. See [enum WindowMode] for possible values and how each mode behaves.
				[b]Note:[/b] Setting the window to fullscreen forcibly sets the borderless flag to [code]true[/code], so make sure to set it back to [code]false[/code] when not wanted.
```

File: doc/classes/DisplayServer.xml [hunk 3]
```diff
@@ -666,6 +667,11 @@
 			<argument index="0" name="mode" type="int" enum="DisplayServer.WindowMode" />
 			<argument index="1" name="window_id" type="int" default="0" />
 			<description>
+				Sets window mode for the given window to [code]mode[/code].
+				See [enum DisplayServer.WindowMode] for possible values and how each mode behaves.
+				
+				[b]Note:[/b] Setting the window to full screen forcibly sets the borderless flag to [code]true[/code], so make sure to set it back to [code]false[/code] when not wanted.
```
Comment (akien-mga) [inline doc/classes/DisplayServer.xml:673]: For the record, I'm not sure that this is intentional.

**Comment by akien-mga** [comment]
Looks good, please squash the commits together, see [PR workflow](https://docs.godotengine.org/en/latest/community/contributing/pr_workflow.html#the-interactive-rebase).

**Comment by DeeJayLSP** [comment]
Done. ~~Took me long time to figure out how to do it~~

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**